ETL_pipeline/pipeline.py

```diff
@@ -3,37 +3,42 @@
 from os import environ as ENV
 
 from dotenv import load_dotenv
+from psycopg2.extensions import connection
 
+from extract import async_api_calls
 from transform import gather_weather_data, gather_air_quality
 from load import (get_db_connection, get_location_id, insert_weather_report,
                   insert_forecast, insert_weather_alert, insert_air_quality)
 
 
-def pipeline(latitude: float, longitude: float) -> None:
+def pipeline(conn: connection, config: dict, latitude: float, longitude: float) -> None:
     """Loads data from weather APIs to a database."""
 
-    load_dotenv()
-
-    weather = gather_weather_data(latitude, longitude)
-    air_quality = gather_air_quality(latitude, longitude, ENV)
+    api_data = async_api_calls(latitude, longitude, config)
 
-    connection = get_db_connection(ENV)
+    weather = gather_weather_data(api_data["weather_for_24hr"],
+                                  api_data["weather_for_week"])
+    air_quality = gather_air_quality(api_data["air_quality"])
 
-    loc_id = get_location_id(connection, latitude, longitude)
-    weather_report_id = insert_weather_report(connection, loc_id)
+    loc_id = get_location_id(conn, latitude, longitude)
+    weather_report_id = insert_weather_report(conn, loc_id)
 
     for w in weather:
-        forecast_id = insert_forecast(
-            connection, w["forecast"], weather_report_id)
+        forecast_id = insert_forecast(conn, w["forecast"], weather_report_id)
```

Comment:
Its better to use something like `w.get('forecast') ` as it is better at error catching if say there is no value